tests/Adapter/CliTest.php

```diff
@@ -31,4 +31,22 @@ public function testException()
         $code = Code::fromString('throw new \Exception("test");');
         $result = $cli->run($code);
     }
+
+    public function testPhpBinary()
+    {
+        $cli = new Cli();
+        $cli->setTempDir(sys_get_temp_dir());
+        $cli->setLogger($this->getMockBuilder('Monolog\Logger')->disableOriginalConstructor()->getMock());
+
+        $code = Code::fromString('return PHP_BINARY;');
+
+        $result = $cli->run($code);
+
+        if ('' === PHP_BINARY) {
+            $this->assertNotEmpty($result);
+            $this->assertFileExists($result);
+        } else {
+            $this->assertEquals(PHP_BINARY, $result);
```

Comment:
```suggestion
            $this->assertSame(PHP_BINARY, $result);
```

`assertSame()` also asserts type.

Reply:
Your absolutely right - done and pushed … cheers!